packages/react-form/src/hooks/test/form-with-dynamic-list.test.tsx

```diff
@@ -142,4 +143,111 @@ describe('useForm with dynamic list', () => {
       expect(isDirty(wrapper)).toBe(false);
     });
   });
+
+  describe('makeCleanAfterSubmit', () => {
+    it('does not undirty fields after successful submit by default', async () => {
+      const promise = Promise.resolve(submitSuccess());
+      const wrapper = mount(
+        <FormWithDynamicVariantList data={fakeProduct()} />,
+      );
+
+      wrapper.find('button', {children: 'Add item'}).trigger('onClick');
+      fillRequiredFields(wrapper);
+
+      expect(isDirty(wrapper)).toBe(true);
+
+      await waitForSubmit(wrapper, promise);
+
+      expect(isDirty(wrapper)).toBe(true);
+    });
+
+    it('does undirty fields after successful submit if makeCleanAfterSubmit is true', async () => {
```

Comment:
I think you meant here that it set dirty to false after submit, right?

Reply:
Yeah! Will change the wording 🙂